data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexType.java

```diff
@@ -1,7 +1,42 @@
 package com.amazon.dataprepper.plugins.sink.opensearch.index;
 
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 public enum IndexType {
-    TRACE_ANALYTICS_RAW,
-    TRACE_ANALYTICS_SERVICE_MAP,
-    CUSTOM;
+    TRACE_ANALYTICS_RAW("trace-analytics-raw"),
+    TRACE_ANALYTICS_SERVICE_MAP("trace-analytics-service-map"),
+    CUSTOM("custom");
+
+    private final String valueString;
+
+    // Reverse-lookup map for getting a IndexType from a name
+    private static final Map<String, IndexType> STRING_TO_INDEX_TYPE_MAP = new HashMap<>();
+
+    static {
+        Arrays.stream(IndexType.values())
+                .forEach(indexType -> STRING_TO_INDEX_TYPE_MAP.put(indexType.valueString, indexType));
+    }
+
+    IndexType(final String name){
+        this.valueString = name;
+    }
+
+    public String getValueString(){
+        return valueString;
+    }
+
+    public static Optional<IndexType> get(final String value) {
```

Comment:
I think this method, and `getIndexTypeValueStrings` can both be package protected.

Reply:
Yes!